runtime/tr.source/trj9/compile/J9Compilation.cpp

```diff
@@ -1193,7 +1193,9 @@ bool
 J9::Compilation::pendingPushLivenessDuringIlgen()
    {
    static bool enabled = (feGetEnv("TR_DisablePendingPushLivenessDuringIlGen") == NULL);
-   return enabled;
+   if (self()->getOSRMode() == TR::involuntaryOSR && self()->getOption(TR_EnableOSROnGuardFailure))
```

Comment:
I'm not certain if this disable should just be limited to involuntaryOSR. OSROnGuardFailure requires the arguments to the call to be placed under the induce, however, OSR liveness does not consider any arguments to the call to be live. Therefore, if OSROnGuardFailure is transitioning to a call with a pending push argument, it will have a dead store before the induce under the current implementation.

This is a bug in the existing liveness implementation which can be resolved by disabling `pendingPushLivenessDuringIlgen` when OSRG is enabled, but the real fix would be to prevent dead stores of any symbols loaded by the induce call.